frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.tsx

```diff
@@ -33,6 +41,41 @@ const resource = {
 
 const CloudShellTerminal: React.FC<CloudShellTerminalProps> = ({ username, onCancel }) => {
   const [data, loaded, loadError] = useK8sWatchResource<CloudShellResource>(resource);
+  const [workSpacePod, setWorkspacePod] = React.useState<InitResponseObject>(null);
+  const [namespace, setNamespace] = React.useState<string>(null);
+  const [apiError, setApiError] = React.useState<string>(null);
+  const [initializing, setInitializing] = React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    if (Array.isArray(data)) {
+      const workspace = data.find(
+        (d) => d?.metadata?.annotations?.[CLOUD_SHELL_USER_ANNOTATION] === username,
+      );
+
+      if (workspace) {
+        const running = workspace.status?.phase === 'Running';
+        if (!initializing && !workSpacePod) {
+          setInitializing(true);
+        }
+        if (!namespace) {
+          setNamespace(workspace.metadata.namespace);
+        }
+        if (running && !apiError && !workSpacePod) {
+          setNamespace(workspace.metadata.namespace);
+          makeTerminalInitCalls(username, workspace.metadata.name, workspace.metadata.namespace)
+            .then((res) => {
+              setWorkspacePod({ pod: res.pod, container: res.container, command: res.cmd });
+              apiError && setApiError(null);
+              initializing && setInitializing(false);
+            })
+            .catch(() => {
+              setApiError("Couldn't reach API");
```

Comment:
Better message would be: `Failed to connect to your OpenShift command line terminal.`

Is there anything in the catch error that's useful to display?